app/javascript/src/NewService/components/ServiceDiscoveryForm.jsx

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import React from 'react'
+import {useState, useEffect} from 'react'
+
+import {FormWrapper, HiddenServiceDiscoveryInput, ErrorMessage,
+  ServiceDiscoveryListItems} from 'NewService/components/FormElements'
+import {CSRFToken, fetchData} from 'utilities/utils'
+import type {FormProps} from 'NewService/types'
+
+const BASE_PATH = '/p/admin/service_discovery'
+const PROJECTS_PATH = `${BASE_PATH}/projects.json`
+
+const ServiceDiscoveryForm = ({formActionPath}: {formActionPath: string}) => {
+  const [projects, setProjects] = useState([])
+  const [services, setServices] = useState([])
+  const [error, setError] = useState({hasError: false, errorMsg: ''})
+
+  const fetchProjects = async () => {
```

Comment:
You can use the same try, catch block to fetch both projects and services. You await for the projects, then await for the services and set the state. If an error happens, you set the error... don't see the use of having 2 separate try, catch blocks.

Reply:
`fetchServices` needs it's own try... catch block.
On loading the component, `fetchProjects` is called, then `fetchServices`. 
But you can call only `fetchServices` when the select changes, in case of having multiple projects

Reply:
This doesn't comes from rails and they may not be fetched never.
Only some users will use this feature, the ones with some OpenShift projects available.
And it will be fetched on user interaction

Reply:
>This doesn't comes from rails and they may not be fetched never.

OK I see.

>And it will be fetched on user interaction

What kind of interaction? It does not depend on any param, only on the radio being selected. So say the user is clicking the radios alternatively many times, would this be fetch every time? Even when the result is constant?
It should also block the UI while the request is ongoing also to tell the user there's some background process working.

Reply:
It will be fetched anytime users clicks. I think it's OK because they may be adding projects in OpenShift, then clicking to fetch again with latest changes.
About blocking the UI or showing something (maybe a spinner), I'd open a new enhancement task.